moveit_core/robot_model/include/moveit/robot_model/joint_model.h

```diff
@@ -139,6 +139,16 @@ class JointModel
    */
   JointModel(const std::string& name, size_t joint_index, size_t first_variable_index);
 
+  /**
+   * @brief      Constructs a joint named \e name with a specified distance factor
+   *
+   * @param[in]  name                   The joint name
+   * @param[in]  index                  The index of the joint in the RobotModel
+   * @param[in]  first_variable_index   The index of the first variable in the RobotModel
+   * @param[in]  distance_factor        The factor applied to the distance between two joint states
+   */
+  JointModel(const std::string& name, size_t joint_index, size_t first_variable_index, double distance_factor);
```

Comment:
I was also wondering between the 2 options, then I chose one that might be more flexible if more arguments are added, although it can be confusing. Do you think that is necessary?

Reply:
I would think that removing the first constructor would be in this case would be ok